src/seth/libexec/seth/seth-call

```diff
@@ -1,29 +1,100 @@
 #!/usr/bin/env bash
 ### seth-call -- call a contract without updating the blockchain
 ### Usage: seth call [<options>] <to> <sig> [<args>]
-###    or: seth call [<options>] <to> <data>
+###    or: seth call [<options>] <to> [<data>]
+###    or: seth call [<options>] --create <code> <sig> [<args>]
+###    or: seth call [<options>] --create <code> [<data>]
 ###
 ### Perform a local call to <to> without publishing a transaction.
 ###
-### With <sig> of the form `<name>(<types>)', infer <data> from <sig>/<args>.
-### With `name(<in-types>)(<out-types>)', also decode the return values.
+### If <sig> has the form `<name>(<types>)', infer <data> from <sig>/<args>.
+### If no <data> is given, just call <to> without any calldata.
+###
+### With `--create', create a new object initialized from <code> and <data>.
+### With `--create' and <sig>, encode constructor args <args> into <data>.
 ###
 ### With `-B <block>', use the state of the blockchain as of <block>.
 ### With `-F <sender>', simulate calling <to> from <sender>
 ### With `-V <value>', simulate transferring <value> to <to>.
 ###
+### With `--hevm`, execute the call locally with hevm
+### With `--code`, overwrite the called contract code
+### With `--debug`, execute the call with hevm's interactive debugger
+###
 ### See also seth-send(1), seth-estimate(1), seth-storage(1), seth-code(1).
 set -e
-[[ $2 ]] || seth --fail-usage "$0"
-TO=$(seth --to-address "$1")
-DATA=$(seth calldata "${@:2}")
-jshon+=(-n {})
-jshon+=(-s "$TO"   -i to)
-jshon+=(-s "$DATA" -i data)
-# shellcheck disable=SC2207
-jshon+=($(seth --send-params))
-jshon+=(-i append)
-[[ $ETH_BLOCK = [0-9]* ]] && ETH_BLOCK=$(seth --to-hex "$ETH_BLOCK")
-jshon+=(-s "${ETH_BLOCK-latest}" -i append)
-result=$(seth rpc eth_call -- "${jshon[@]}")
-seth --abi-decode "$2" "$result"
+
+[[ $1 ]] || seth --fail-usage "$0"
+if [[ $SETH_CREATE ]]; then
+  DATA=$(seth --to-hexdata "$1")
+  if [[ $2 ]]; then
+    args=$(seth calldata "${@:2}")
+    args=${args#0x}
+    DATA+=${args:8}
+  fi
+else
+  TO=$(seth --to-address "$1")
+  if [[ $2 ]]; then
+    DATA=$(seth calldata "${@:2}")
+  fi
+fi
+
+if [[ -z "$HEVM_DEBUG" && -z "$HEVM_EXEC" ]]; then
+  jshon+=(-n {})
+  [[ $TO ]] && jshon+=(-s "$TO" -i to)
+  [[ $DATA ]] && jshon+=(-s "$DATA" -i data)
+  # shellcheck disable=SC2207
+  jshon+=($(seth --send-params))
+  jshon+=(-i append)
+  [[ $ETH_BLOCK = [0-9]* ]] && ETH_BLOCK=$(seth --to-hex "$ETH_BLOCK")
+  jshon+=(-s "${ETH_BLOCK-latest}" -i append)
+  result=$(seth rpc eth_call -- "${jshon[@]}")
+  seth --abi-decode "$2" "$result"
+else
+  opts=()
+  [[ "$HEVM_DEBUG" ]] && opts+=(--debug)
+  [[ "$HEVM_STATE" ]] && opts+=(--state "$HEVM_STATE")
+
+  opts+=(--rpc "$ETH_RPC_URL")
```

Comment:
in other places this defaults to a local node when not defined, see https://github.com/dapphub/dapptools/pull/446/files#diff-d3a1496555063b19b8bdb94c3476de0aR37

Reply:
Will fix. We should really set all these defaults at the top level imo.